determined-ai/determined
# fix: Save experiment glide table selection rows in settings [WEB-1366]

## Description

Stores selectAll, selectedExperimentIds, and excludedExperimentIds in persistent settings, so you can leave the new experiment list page and come back with same settings.  These will be neutral values while waiting for `isLoadingSettings`

Previously the source of truth for selection was the GlideTable and data[index] instead of experiment.id, so there are a few changes.
I kept `excludedExperimentIds` as a Set, though it's stored to settings as an array; I am uncertain about whether to keep this a Set and `selectedExperimentIds` being an Array

## Test Plan

**Test on a single page of experiments (latest-main project 107) and a multi-page experiment list (uncategorized)**

Selection
- With `?f_explist_v2=on`, select every other experiment in a list. Refresh or navigate away from the page and return; same experiments should be selected
- The count of selected experiments in the edit button, e.g. "Edit (10)" should reflect the count selected

Select All
- Check the header checkbox on the table - this should select all remaining experiments
- Reload the page; this should keep select-all state, experiments should all be selected, "Edit (10)" should reflect count

Exclusion
- While all are selected, uncheck one row to exclude it.
- Reload the page; this should keep the select-all state, and all should be selected except for the excluded; "Edit (9)" should reflect count


## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
